### PR TITLE
Allow to specify controller and argocd namespaces

### PR DESCRIPTION
**Pull request by maruina**
Update the chart to support the latest command options and `appVersion`.

Blocked by #130 and #132 .

### CODE REVIEW

File: charts/applicationset-progressive-sync/templates/deployment.yaml [hunk 1]
```diff
@@ -48,6 +48,8 @@ spec:
             {{- if or (gt ( .Values.replicaCount | int64) 1) .Values.args.enableLeaderElection }}
             - --enable-leader-election=true
             {{- end }}
+            - --namespace={{ .Values.args.Namespace }}
+            - --argo-namespace={{ .Values.args.argoNamespace }}
```
Comment (sledigabel) [inline charts/applicationset-progressive-sync/templates/deployment.yaml:52]: perhaps defaulting to the same? (if empty, fill it with the namespace?

File: charts/applicationset-progressive-sync/README.md [hunk 2]
```diff
@@ -26,8 +26,10 @@ Users of Helm v2 should use `--set installCRDs=true` to install the CRDs.
 | Key | Type | Default | Description |
 |-----|------|---------|-------------|
 | affinity | object | `{}` |  |
+| args.argoNamespace | string | `"argocd"` |  |
```
Comment (sledigabel) [inline charts/applicationset-progressive-sync/README.md:29]: we should leverage the builtin `Release.Namespace`?
https://helm.sh/docs/chart_template_guide/builtin_objects/

File: charts/applicationset-progressive-sync/README.md [hunk 3]
```diff
@@ -26,8 +26,10 @@ Users of Helm v2 should use `--set installCRDs=true` to install the CRDs.
...
 | args.enableLeaderElection | bool | `false` |  |
 | args.metricsAddr | string | `":8080"` |  |
+| args.namespace | string | `"argocd"` |  |
```
Comment (sledigabel) [inline charts/applicationset-progressive-sync/README.md:32]: we should leverage the builtin Release.Namespace?
https://helm.sh/docs/chart_template_guide/builtin_objects/

Also We should add some documentation

File: charts/applicationset-progressive-sync/templates/deployment.yaml (hunk 1 above)
Comment (maruina) [reply charts/applicationset-progressive-sync/templates/deployment.yaml:52]: They already default to the same, but I think I'm missing your suggestion.

File: charts/applicationset-progressive-sync/README.md (hunk 3 above)
Comment (maruina) [reply charts/applicationset-progressive-sync/README.md:32]: I don't think we can use an helm object as value. Again, I think I'm missing your suggestion

File: charts/applicationset-progressive-sync/templates/deployment.yaml (hunk 1 above)
Comment (sledigabel) [reply charts/applicationset-progressive-sync/templates/deployment.yaml:52]: Ok I'll add some suggestions instead

File: charts/applicationset-progressive-sync/README.md [hunk 4]
```diff
@@ -26,8 +26,10 @@ Users of Helm v2 should use `--set installCRDs=true` to install the CRDs.
 | Key | Type | Default | Description |
 |-----|------|---------|-------------|
 | affinity | object | `{}` |  |
-| args.enableLeaderElection | bool | `false` |  |
-| args.metricsAddr | string | `":8080"` |  |
+| args.argoNamespace | string | `"argocd"` | The namespace where ArgoCD and the ApplicationSet controller are deployed to. |
+| args.enableLeaderElection | bool | `false` | Enable leader election for controller manager. |
+| args.metricsAddr | string | `":8080"` | The address the metric endpoint binds to. |
+| args.namespace | string | `"argocd"` | The controller namespace. |
```
Comment (sledigabel) [inline charts/applicationset-progressive-sync/README.md:32]: I don't think we need it

File: charts/applicationset-progressive-sync/templates/deployment.yaml [hunk 5]
```diff
@@ -48,6 +48,8 @@ spec:
             {{- if or (gt ( .Values.replicaCount | int64) 1) .Values.args.enableLeaderElection }}
             - --enable-leader-election=true
             {{- end }}
+            - --namespace={{ .Values.args.Namespace }}
```
Comment (sledigabel) [inline charts/applicationset-progressive-sync/templates/deployment.yaml:51]:
```suggestion
            - --namespace={{ .Release.Namespace }}
```

File: charts/applicationset-progressive-sync/templates/deployment.yaml (hunk 1 above)
Comment (sledigabel) [inline charts/applicationset-progressive-sync/templates/deployment.yaml:52]:
```suggestion
            - --argo-namespace={{ .Values.args.argoNamespace | defaults .Release.Namespace }}
```

File: charts/applicationset-progressive-sync/values.yaml [hunk 6]
```diff
@@ -13,8 +13,14 @@ image:
   tag: ""
 
 args:
+  # -- The address the metric endpoint binds to.
   metricsAddr: :8080
+  # -- Enable leader election for controller manager.
   enableLeaderElection: false
+  # -- The controller namespace.
+  namespace: argocd
```
Comment (sledigabel) [inline charts/applicationset-progressive-sync/values.yaml:21]: I don't think we need this one.

File: charts/applicationset-progressive-sync/templates/deployment.yaml (hunk 1 above)
Comment (maruina) [reply charts/applicationset-progressive-sync/templates/deployment.yaml:52]: Well at this point shouldn't just be `{{ .Release.Namespace }}`?

Comment (maruina) [reply charts/applicationset-progressive-sync/templates/deployment.yaml:52]: Pushed

File: charts/applicationset-progressive-sync/README.md (hunk 4 above)
Comment (maruina) [reply charts/applicationset-progressive-sync/README.md]: Fixed

File: charts/applicationset-progressive-sync/values.yaml (hunk 6 above)
Comment (maruina) [reply charts/applicationset-progressive-sync/values.yaml]: Fixed

**Review by nebojsa-prodana — APPROVED** [review]
LGTM! Allows controller to be deployed to a different namespace from argocd while still being able to watch on argocd resources.

**Pull request merged**